lib/phonebook/providers/association_list_provider.dart

```diff
@@ -12,8 +12,21 @@ class AssociationListNotifier extends ListNotifier<Association> {
     associationRepository.setToken(token);
   }
 
-  Future<AsyncValue<List<Association>>> loadAssociations([String? filter]) async {
-    return await loadList(() async => associationRepository.getAssociationList(filter!));
+  late List<Association> associationList;
+
+  Future<AsyncValue<List<Association>>> loadAssociations() async {
+    associationList = await associationRepository.getAssociationList();
+    return await loadList(() async => associationRepository.getAssociationList());
```

Comment:
You can mock the API response (for testing purposes)

`return await loadList(() async => [ Youre mocked data here]);`

The same goes for the other function that involve repository function call